llvm/lib/Target/AIE/AIEPHISorterPass.cpp

```diff
@@ -0,0 +1,101 @@
+//==========-- AIEPHISorterPass.cpp - Sort PHI nodes in loops --==============//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+//
+// This Pass reorders PHI nodes in a more friendly way to Greedy Allocator
+//
+//===----------------------------------------------------------------------===//
+
+#include "AIE2.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Analysis/LoopInfo.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/IntrinsicInst.h"
+#include "llvm/IR/IntrinsicsAArch64.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/InitializePasses.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Transforms/Utils/Cloning.h"
```

Comment:
I think we only need a fraction of these includes. And as remarked, I think it could be independent on LoopInfo. If you want to generalize later, I would already make the 'is part of the body' test based on comparing getLoopFor of both blocks for equality.